### DISCUSSION TITLE
Making OSU more touch/mobile friendly

### DISCUSSION THREAD

**Discussion by jimbot03** [ppy/osu#18050] [Ideas] [answered]
Understandably the main focus of development is still the desktop clients and that's completely fine 

However I'm gonna be a little selfish and say that when I'm out and about sometimes I get an itch to play a rhythm game and for me that's OSU because it's the game that got me into rhythm games. That said in its current state its not very usable on mobile and to be clear that is no fault of the devs and has everything to do with mapping touch controls on mostly small displays while maintaining the ability to tap 

With all that said I'm going to list of couple QOL improvements that would make mobile more useable and user friendly

Tapping consistency --
   
   This is the biggest issue I see currently although OSU lazer does support multi touch in game every touch is counted as a M1 click 
   and as a result tapping in quick succession is nearly impossible as the game wont let you click M1 while M1 is already held down 
   this is compounded by the more sluggish nature of tapping on a screen rather then using keyboard strokes and makes even 
   moderately slow bpm maps almost unplayable on touch alone 

   I see two solutions for this one of them is to alternate each consecutive touchscreen tap between M1 and M2 while not a perfect 
   solution it would be relatively easy to implement and make things far more playable 

   As for the second solution this one is a bit more dumb but adding a toggle to the settings to allow half the display and/or a 
   dedicated segment to be used for M1 and M2 separately on mobile this would be useless for standalone playing however with a 
   mouse connected or playing with autopilot would work wonders the downside being it's more effort to implement  

Mouse Smoothing/Acceleration on android -

   This one isn't that big a deal because using mouse on android is certainly a niche category of use and I suspect this would be a pain 
   to solve but the title speaks for itself on most android devices there is no way to use raw input on connected mouses I have no idea 
   if this is fixable with backend code but for obvious reasons this makes using mouse for tracking mostly unviable on android 

Importing beatmaps from desktop to mobile -
  
   So this one is quite simple as someone who has a sizable collection of beatmaps on the desktops clients it would be nice to have a 
   easier way to import already downloaded beatmaps to android 

   I'm not sure how to go about a solution for this one with my one idea being making it so that android can see and import a copy of 
   my stable client beatmaps if I copy them to the device  

Default UI scaling - 

   This one is just a minor gripe and honestly probably nit picking but on smaller mobile displays the UI can be too small to click 
   consistently now obviously that's because the default UI scaling is meant for desktop that said it should be quite easy to default it 
   scaled larger for mobile devices 

And that's it 
I'm well aware that this will come off as a lot I struggle to simplify my thoughts so it tends to come out as text walls
That said I want to reiterate that these are suggestions and not demands and any feedback or insight is welcome

**Comment by bdach** [comment, marked as answer]
We know about most of this, most of these already have discussion threads. Mobile work will happen when there's bandwidth for it.
